integration-libs/cdp/root/components/cdp-order/cdp-order.component.ts

```diff
@@ -9,120 +9,101 @@ import {
+import { returnOrder } from '../../model/returnDetail/returnOrder';
+import { CdpOrderService } from '../../service';
 
 @Component({
   selector: 'cx-order',
   templateUrl: './cdp-order.component.html',
   styleUrls: ['./cdp-order.component.scss'],
-  providers: [CxDatePipe],
+  providers: [CxDatePipe,CdpOrderService],
 })
-export class OrderComponent implements OnInit {
+export class cdpOrderComponent implements OnInit {
   constructor(
-    private userIdService: UserIdService,
-    private cdpOrderAdapter: cdpOrderAdapter,
     protected datePipe: CxDatePipe,
     protected routing: RoutingService,
     protected occEndpointsService: OccEndpointsService,
-    protected translation: TranslationService
+    protected translation: TranslationService,
+    protected cdpOrderService: CdpOrderService,
   ) {}
 
-  result: finalOrder = { orders: [] };
+  orderValue: finalOrder = { orders: [] };
   totalPrice: number = 0;
   totalItem: number[] = [];
   orderDetail: Record<string, order> = {};
-  i: number = 0;
-  output: result;
   orderStatus: Record<string, Record<string, number>> = {};
   orderImage: Record<string, product[]> = {};
+  returnDate: Record<string, string>={};
   userId: string;
   tabTitleParam$ = new BehaviorSubject(0);
   public loading$ = new BehaviorSubject<boolean>(true);
   sortType: string;
+  obser$: Observable<finalOrder>;
+  returnObser$: Observable<returnOrder>;
+  orderReturn: returnOrder;
+  page_size: number = 5;
 
   ngOnInit(): void {
     this.getMyData();
   }
 
   public getMyData(): void {
-    const obser$ = this.userIdService
-      .takeUserId()
-      .pipe(switchMap((userId) => this.cdpOrderAdapter.getOrder(userId)));
-
-    obser$.subscribe((res) => {
-      this.result = res;
+    this.obser$ = this.cdpOrderService.getOrder(this.page_size);
+    this.obser$.subscribe((res) => {
+      this.orderValue = res;
       this.tabTitleParam$.next(res.orders.length);
-      this.calculateTotalAmount(this.result);
-      this.getItemCount(this.result);
-      console.log(this.result);
+      this.calculateTotalAmount(this.orderValue);
+      this.getItemCount(this.orderValue);
+      this.fetchReturn();
     });
   }
 
   public calculateTotalAmount(finalResult: finalOrder): void {
     for (var val of finalResult.orders) {
       this.totalPrice = val.total.value + this.totalPrice;
-      console.log(this.totalPrice);
     }
   }
 
   public async getItemCount(finalResult: finalOrder): Promise<void> {
-    for (let orderval of finalResult.orders) {
-      await this.userIdService
-        .takeUserId()
-        .pipe(
-          mergeMap((userId) =>
-            this.cdpOrderAdapter.getOrderDetail(userId, orderval)
-          )
-        )
-        .toPromise()
-        .then((data) => {
-          this.orderDetail[orderval.code] = data;
-        });
-    }
+    await this.cdpOrderService.fetchOrderDetail(finalResult).then((data)=>{
+      this.orderDetail= data;
```

Comment:
You can populate the image and product details by calling the relevant methods using Observables.